### PR TITLE
[CSL-2201] Rebase `feature/new-wallet` on top of `develop`

### PR DESCRIPTION
**Pull request by sectore**
_Note:_  **Don't merge** - this PR was needed to check commits and to discuss changes.

_Task:_ https://iohk.myjetbrains.com/youtrack/issue/CSL-2201

_**Update (24/01/18):**_ All changes / commits of this PR have been landed in branch [`feature/new-wallet`](https://github.com/input-output-hk/cardano-sl/tree/feature/new-wallet).

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -91,21 +91,13 @@ test_script:
...
-  - copy cardano-node.exe daedalus\
-  # Build the bridge
-  - cd daedalus
-  - Echo %APPVEYOR_BUILD_VERSION% > build-id
-  - Echo %APPVEYOR_REPO_COMMIT% > commit-id
-  - Echo https://ci.appveyor.com/project/%APPVEYOR_ACCOUNT_NAME%/%APPVEYOR_PROJECT_SLUG%/build/%APPVEYOR_BUILD_VERSION% > ci-url
-  - ps: Install-Product node 7
-  - ..\scripts\ci\appveyor-retry call npm install
-  - npm run build:prod
+  - copy cardano-node.exe daedalus
```
Comment (adinapoli-iohk) [inline appveyor.yml]: I really cannot judge if the following is correct. Maybe let's not bother now and when we open a PR from `feature/new-wallet` into `develop` we ask devops to review everything.

File: appveyor.yml [hunk 2]
```diff
@@ -116,7 +108,7 @@ after_test:
       }
       .\cache-s3 --prefix=$env:APPVEYOR_PROJECT_NAME --git-branch=$env:APPVEYOR_REPO_BRANCH --suffix=windows -c -v info save stack work
     }
- - xcopy /q /s /e /r /k /i /v /h /y "%WORK_DIR%\daedalus" "%APPVEYOR_BUILD_FOLDER%\daedalus"
+# - xcopy /q /s /e /r /k /i /v /h /y "%WORK_DIR%\daedalus" "%APPVEYOR_BUILD_FOLDER%\daedalus"
```
Comment (adinapoli-iohk) [inline appveyor.yml]: Ditto.

File: daedalus/.babelrc [hunk 3]
```diff
@@ -1,6 +0,0 @@
-{
```
Comment (adinapoli-iohk) [inline daedalus/.babelrc]: The above binary file looks like it was committed by accident, we should remove it. ^^

File: explorer/frontend/nixpkgs [hunk 4]
```diff
@@ -0,0 +1 @@
+/nix/store/mz8k9ldyi6bz5wdq6xpmf3s3vqw3w5mn-61fbdb47a69f78998f55207e64122a0798047b5d.tar.gz-unpacked
```
Comment (adinapoli-iohk) [inline explorer/frontend/nixpkgs]: Are we sure this file is meant to be there? My hunch is that it was committed by mistake.

File: scripts/check_deps.sh [hunk 5]
```diff
@@ -27,4 +25,5 @@ echo $total_checksum
 # Ignore whitespaces
 if [ "$(echo "$total_checksum" | tr -d '[:space:]')" != "$(echo "$existing_checksum" | tr -d '[:space:]')" ]; then
   echo "ERROR: you need to (run ./pkgs/generate.sh) and commit the changes."
+  echo "$total_checksum" > ./scripts/deps_checksum
```
Comment (adinapoli-iohk) [inline scripts/check_deps.sh]: Do we know why our feature branch changes this file? I think we should just "Copy and paste" the revision `develop` is currently using, so the merge is easier.

File: scripts/ci/travis.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+export BUILD_UID="$TRAVIS_OS_NAME-${TRAVIS_BRANCH//\//-}"
+export XZ_OPT=-1
+
+# For now we dont have macOS developers on explorer
+if [[ ("$TRAVIS_OS_NAME" == "linux") ]]; then
+  # Generate explorer frontend
+  EXPLORER_EXECUTABLE=$(pwd)/cardano-sl-explorer-static.root/bin/cardano-explorer-hs2purs ./explorer/frontend/scripts/build.sh
+
+  echo "Packing up explorer-frontend ..."
+  tar cJf s3/explorer-frontend-$BUILD_UID.tar.xz explorer/frontend/dist
```
Comment (adinapoli-iohk) [inline scripts/ci/travis.sh]: I think this shouldn't be here, we are not using Travis anymore.

File: wallet/purescript/Main.hs [hunk 7]
```diff
@@ -1,109 +0,0 @@
-module Main
```
Comment (adinapoli-iohk) [inline wallet/purescript/Main.hs]: Spurious bin file above ^^

File: wallet/src/Pos/Wallet/Web/Methods/Logic.hs [hunk 8]
```diff
@@ -9,6 +9,7 @@ module Pos.Wallet.Web.Methods.Logic
        , getWallet
        , getWallets
        , getAccount
+       , getAccountMod
```
Comment (adinapoli-iohk) [inline wallet/src/Pos/Wallet/Web/Methods/Logic.hs]: Let's check this export is needed anywhere else. If not, let's scrap it.

File: scripts/check_deps.sh (hunk 5 above)
Comment (ksaric) [reply scripts/check_deps.sh]: I changed it because the script wasn't detecting what the current state of dependencies are. It does now.

Comment (adinapoli-iohk) [reply scripts/check_deps.sh]: Great, in which case let's just move to the revision in `develop`. Less merge pain 😉

File: daedalus/.babelrc (hunk 3 above)
Comment (sectore) [reply daedalus/.babelrc]: Please check your comment ^ again, `daedalus/.babelrc` has been removed w/ this PR.

File: wallet/purescript/Main.hs (hunk 7 above)
Comment (sectore) [reply wallet/purescript/Main.hs]: That's why it has been removed by this PR ;)

File: daedalus/.babelrc (hunk 3 above)
Comment (adinapoli-iohk) [reply daedalus/.babelrc]: I couldn't attach a comment to a binary diff, so that comment referred to that `cabal-config-dist` which is not gone it seems 😉

**Comment by sectore** [comment]
All changes / commits above ^ have been landed in branch [`feature/new-wallet`](https://github.com/input-output-hk/cardano-sl/tree/feature/new-wallet).

**Pull request closed without merging**